File: BotsSDK/korebotsdklib/src/main/java/kore/botssdk/models/BrandingBodyTimeStampModel.java
package kore.botssdk.models;

import java.io.Serializable;

import kore.botssdk.utils.StringUtils;

public class BrandingBodyTimeStampModel implements Serializable {
    private Boolean show;
    private String show_type;
    private String position;
    private String separator;
    private String color;

    public BrandingBodyTimeStampModel updateWith(BrandingBodyTimeStampModel configModel) {
        show = configModel.show != null ? configModel.show : show;
        show_type = !StringUtils.isNullOrEmpty(configModel.show_type) ? configModel.show_type : show_type;
        position = !StringUtils.isNullOrEmpty(configModel.position) ? configModel.position : position;
        separator = !StringUtils.isNullOrEmpty(configModel.separator) ? configModel.separator : separator;
        color = !StringUtils.isNullOrEmpty(configModel.color) ? configModel.color : color;
        return this;
    }

    public boolean isShow() {
        return show != null ? show : false;
    }

    public String getSeparator() {
        return separator;
    }

    public void setSeparator(String separator) {
        this.separator = separator;
    }

    public String getColor() {
        return color;
    }

    public String getPosition() {
        return position;
    }

    public String getShow_type() {
        return show_type;
    }

    public void setColor(String color) {
        this.color = color;
    }

    public void setShow(boolean show) {
        this.show = show;
    }

    public void setPosition(String position) {
        this.position = position;
    }

    public void setShow_type(String show_type) {
        this.show_type = show_type;
    }
}

File: BotSdk/botclient/src/main/java/com/kore/network/api/responsemodels/branding/BrandingBodyTimeStampModel.kt
package com.kore.network.api.responsemodels.branding

import com.google.gson.annotations.SerializedName

data class BrandingBodyTimeStampModel(
    @SerializedName("show") val show: Boolean?,
    @SerializedName("show_type") val showType: String = "",
    @SerializedName("position") val position: String = "",
    @SerializedName("separator") val separator: String = "",
    @SerializedName("color") val color: String = "",
    @SerializedName("time_format") val timeFormat: String = "12",
    @SerializedName("date_format") val dateFormat: String = "",
)

fun BrandingBodyTimeStampModel.updateWith(configModel: BrandingBodyTimeStampModel?): BrandingBodyTimeStampModel {
    return this.copy(
        show = configModel?.show ?: show,
        showType = configModel?.showType?.ifEmpty { this.showType } ?: showType,
        position = configModel?.position?.ifEmpty { this.position } ?: position,
        separator = configModel?.separator?.ifEmpty { this.separator } ?: separator,
        color = configModel?.color?.ifEmpty { this.color } ?: color,
        timeFormat = configModel?.timeFormat?.ifEmpty { timeFormat } ?: timeFormat,
        dateFormat = configModel?.dateFormat?.ifEmpty { this.dateFormat } ?: dateFormat,
    )
}